ocrd_tesserocr/deskew.py

```diff
@@ -0,0 +1,216 @@
+from __future__ import absolute_import
+
+import os.path
+import math
+from tesserocr import (
+    PyTessBaseAPI,
+    PSM, OEM,
+    Orientation,
+    WritingDirection,
+    TextlineOrder
+)
+
+from ocrd_utils import (
+    getLogger, concat_padded,
+    MIMETYPE_PAGE
+)
+from ocrd_modelfactory import page_from_file
+from ocrd_models.ocrd_page import (
+    MetadataItemType,
+    LabelsType, LabelType,
+    AlternativeImageType,
+    TextRegionType, PageType,
+    to_xml
+)
+from ocrd import Processor
+
+from .config import TESSDATA_PREFIX, OCRD_TOOL
+from .common import (
+    image_from_page,
+    image_from_region,
+    save_image_file,
+    membername
+)
+
+TOOL = 'ocrd-tesserocr-deskew'
+LOG = getLogger('processor.TesserocrDeskew')
+FILEGRP_IMG = 'OCR-D-IMG-DESKEW'
+
+class TesserocrDeskew(Processor):
+
+    def __init__(self, *args, **kwargs):
+        kwargs['ocrd_tool'] = OCRD_TOOL['tools'][TOOL]
+        kwargs['version'] = OCRD_TOOL['version']
+        super(TesserocrDeskew, self).__init__(*args, **kwargs)
+
+    def process(self):
+        """Performs region-level deskewing with Tesseract on the workspace.
+        
+        Open and deserialise PAGE input files and their respective images,
+        then iterate over the element hierarchy down to the region level
+        for all text and table regions.
+        
+        Set up Tesseract to recognise the region image's orientation, skew
+        and script (with both OSD and AnalyseLayout). Rotate the image
+        accordingly, and annotate the angle, readingDirection and textlineOrder.
+        
+        Create a cropped (and possibly deskewed) image file, and reference it
```

Comment:
Is cropping mandatory from a tessapi point-of-view?

Reply:
The docstring was misleading. I changed "cropped" to "corresponding"